apps/framework-cli/src/cli/local_webserver.rs

```diff
@@ -515,6 +530,35 @@ async fn handle_json_array_body(
     success_response(url)
 }
 
+async fn validate_token(token: Option<&str>, env_var: &str) -> bool {
+    match env::var(env_var) {
+        Ok(ingest_api_key) => match token {
+            Some(token) => hex_digest(Algorithm::SHA256, token.as_bytes()) == ingest_api_key,
```

Comment:
We should not be using SHA-256 for this.

NIST recommends PBKDF2 - let's follow that recommendation with a constant time check on the output.

Ref: https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#pbkdf2

https://github.com/RustCrypto/password-hashes/tree/master/pbkdf2